mantle/cmd/ore/azure/upload-blob-arm.go

```diff
@@ -97,16 +97,20 @@ func runUploadBlobARM(cmd *cobra.Command, args []string) {
 		plog.Fatalf("No storage service keys found")
 	}
 
-	for _, k := range *kr.Keys {
-		if err := api.UploadBlob(ubo.storageacct, *k.Value, ubo.vhd, ubo.container, ubo.blob, ubo.overwrite); err != nil {
-			plog.Fatalf("Uploading blob failed: %v", err)
-		}
-		break
+	//only use the first service key to avoid uploading twice
+	//see https://github.com/coreos/coreos-assembler/pull/1849
+	k := (*kr.Keys)[0]
```

Comment:
You might need to update the `kr.Keys == nil` check to also make sure that there is at least one key in `kr.Keys` now.

Reply:
Make sense. How about:
```
if kr.Keys == nil || len(*kr.Keys) == 0 {
		plog.Fatalf("No storage service keys found")
	}
```